packages/utils/rollup.config.ts

```diff
@@ -1,36 +1,50 @@
+import typescript from "@rollup/plugin-typescript"
 import * as fs from "fs"
 import cleanup from "rollup-plugin-cleanup"
-import typescript from "rollup-plugin-typescript2"
 
 const pkg = JSON.parse(fs.readFileSync("./package.json", "utf-8"))
 const banner = `/**
  * @module ${pkg.name}
  * @version ${pkg.version}
  * @file ${pkg.description}
- * @copyright Ethereum Foundation 2024
+ * @copyright Ethereum Foundation ${new Date().getFullYear()}
  * @license ${pkg.license}
  * @see [Github]{@link ${pkg.homepage}}
 */`
 
-export default {
-    input: "src/index.ts",
-    output: [
-        { file: pkg.exports["."].require, format: "cjs", banner, exports: "auto" },
-        { file: pkg.exports["."].default, format: "es", banner },
-        {
-            dir: "./dist/lib.commonjs",
-            format: "cjs",
-            banner,
-            preserveModules: true,
-            entryFileNames: "[name].cjs"
-        },
-        { dir: "./dist/lib.esm", format: "es", banner, preserveModules: true }
-    ],
-    plugins: [
-        typescript({
-            tsconfig: "./build.tsconfig.json",
-            useTsconfigDeclarationDir: true
-        }),
-        cleanup({ comments: "jsdoc" })
-    ]
-}
+export default [
+    {
+        input: "src/index.ts",
+        output: [
+            { file: pkg.exports["."].require, format: "cjs", banner, exports: "auto" },
+            { file: pkg.exports["."].default, format: "es", banner }
+        ],
+        plugins: [
+            typescript({
+                tsconfig: "./build.tsconfig.json"
+            }),
+            cleanup({ comments: "jsdoc" })
+        ]
+    },
+    {
+        input: "src/index.ts",
+        output: [
+            {
+                dir: "./dist/lib.commonjs",
+                format: "cjs",
+                banner,
+                preserveModules: true,
+                entryFileNames: "[name].cjs"
+            },
+            { dir: "./dist/lib.esm", format: "es", banner, preserveModules: true }
+        ],
+        plugins: [
+            typescript({
+                tsconfig: "./build.tsconfig.json",
+                declaration: false,
```

Comment:
Are `declaration ` and `declarationDir ` necessary here?

Reply:
Yes, we need to prevent the Rollup typescript plugin from generating types in that configuration, otherwise it will throw an error.